src/libraries/System.Private.CoreLib/src/System/Globalization/IdnMapping.Nls.cs

```diff
@@ -104,17 +104,6 @@ private unsafe string GetUnicodeCore(string asciiString, char* ascii, int count,
         // ---- PAL layer ends here ----
         // -----------------------------
 
-        private uint Flags
-        {
-            get
-            {
-                int flags =
-                    (AllowUnassigned ? Interop.Normaliz.IDN_ALLOW_UNASSIGNED : 0) |
```

Comment:
I see that these System.Globalization.Native flags happen to have the same values as the Windows NLS flags that allowed you to unify this; but I think it makes the code less readable. I would keep the flags and this method duplicated.

Reply:
The reason why I did this was to avoid declaring a `NlsFlags` and an `IcuFlags` property. I'm curious, why do you think it makes code less readable?

Reply:
It is not obvious that this constant is actually defined by Windows SDK, with a different name (https://docs.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-idntoascii). The coding convention is to use the platform names for constants defined by the platform.

I think it would be fine to have both `NlsFlags` and `IcuFlags` properties.

Reply:
Yeah, I already changed it, I was curious why you thought it was less readable.